src/transformers/generation_utils.py

```diff
@@ -1042,14 +1463,35 @@ def beam_search(
         while cur_len < max_length:
             model_inputs = self.prepare_inputs_for_generation(input_ids, **model_kwargs)
 
-            outputs = self(**model_inputs, return_dict=True)
+            outputs = self(
+                **model_inputs,
+                return_dict=True,
+                output_attentions=output_attentions,
+                output_hidden_states=output_hidden_states,
+            )
             next_token_logits = outputs.logits[:, -1, :]
 
             # adjust tokens for Bart, *e.g.*
             next_token_logits = self.adjust_logits_during_generation(
                 next_token_logits, cur_len=cur_len, max_length=max_length
             )
 
+            # Store logits, attentions and hidden_states when required
+            if return_dict_in_generate:
+                if output_scores:
+                    logits += (next_token_logits,)
```

Comment:
Here we are interested in the `next_token_scores` after this line `next_token_scores = next_token_scores + beam_scores[:, None].expand_as(next_token_scores)`